packages/middleware-sdk-s3-control/src/prepend-account-id.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  BuildHandler,
+  BuildHandlerArguments,
+  BuildHandlerOptions,
+  BuildHandlerOutput,
+  BuildMiddleware,
+  MetadataBearer,
+  Pluggable
+} from "@aws-sdk/types";
+import { HttpRequest } from "@aws-sdk/protocol-http";
+
+export function prependAccountIdMiddleware(): BuildMiddleware<any, any> {
+  return <Output extends MetadataBearer>(
+    next: BuildHandler<any, Output>
+  ): BuildHandler<any, Output> => async (
+    args: BuildHandlerArguments<any>
+  ): Promise<BuildHandlerOutput<Output>> => {
+    let { request } = args;
+    const { input } = args;
+    if (HttpRequest.isInstance(request) && input.AccountId) {
+      const newHostname = input.AccountId + "." + request.hostname;
+      request.hostname = newHostname;
+      delete request.headers["x-amz-account-id"];
+    }
+    return next({
+      ...args,
+      request
+    });
+  };
+}
+
+export const prependAccountIdMiddlewareOptions: BuildHandlerOptions = {
+  step: "build",
+  tags: ["PREPEND_ACCOUNT_ID_MIDDLEWARE"],
+  name: "prependAccountIdMiddleware"
+};
```

Comment:
Since we update the `hostname` here, do we need to make it before `hostHeaderMiddleware` like in [bucketEndponitMiddleware](https://github.com/aws/aws-sdk-js-v3/blob/9e81bd4fe5df3a28639b6e5abfe4fc652022e488/packages/middleware-bucket-endpoint/src/bucketEndpointMiddleware.ts#L55-L62)?

Reply:
fixed in b736599a4075b7f232a6ce4ff56767d4f077b4ac